lib/cfn-nag/custom_rules/password_base_rule.rb

```diff
@@ -36,7 +36,9 @@ def audit_impl(cfn_model)
 private
 
 def verify_parameter_exists(resource, password_property, sub_property_name)
-  if sub_property_name.nil?
+  if resource.send(password_property).nil?
+    true
```

Comment:
i read this as:
Given the password_property does not exist on the resource
Then the "parameter" exists

So i'm wondering if there are two naming aspects we need to fix:
1. "parameter" is really "password_property"?
2. The other names might be confusing me, but is it really a matter of existence?  If the password_property returns a nil value, I guess I'd expect "exists?" predicate to be false?

Reply:
1. I think I can tweak the naming here for the parameter/sub_property_name to work a bit better and read more clearly. The original setup was to just have the first level parameter from `Properties`, but there are several "parameters" now that are in a sub level even further down. 

2. I see the confusing naming and results now that you point that out. I need to address this to read more clearly so it makes more sense from a quick glance.